app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -137,6 +136,183 @@ object DialogUtils {
+                huePicker.setHue(it.toString().toInt())
+                forceViewUpdates(calculatedColorState)
+            }
+        }
+        satText.addTextChangedListener {
+            if (isIntegerInput(it.toString())) {
+                val sat = (it.toString().toFloat() / 100f).coerceIn(0f, 1f)
+                val hsv = FloatArray(3)
+                hsv[0] = (hueText.text.toString().toFloat())
+                hsv[1] = sat
+                hsv[2] = (valText.text.toString().toFloat() / 100f).coerceIn(0f, 1f)
+                val calculatedColorState = ColorUtils.calculateColorWithHSV(hsv)
+                forceViewUpdates(calculatedColorState)
+            }
+        }
+        valText.addTextChangedListener {
+            if (isIntegerInput(it.toString())) {
+                val value = (it.toString().toFloat() / 100f).coerceIn(0f, 1f)
+                val hsv = FloatArray(3)
+                hsv[0] = (hueText.text.toString().toFloat())
+                hsv[1] = (satText.text.toString().toFloat() / 100f).coerceIn(0f, 1f)
+                hsv[2] = value
+                val calculatedColorState = ColorUtils.calculateColorWithHSV(hsv)
+                forceViewUpdates(calculatedColorState)
+            }
+        }
+        redText.addTextChangedListener {
+            if (isIntegerInput(it.toString())) {
+                val rgb = IntArray(3)
+                rgb[0] = it.toString().toInt().coerceIn(0, 255)
+                rgb[1] = greenText.text.toString().toInt().coerceIn(0, 255)
+                rgb[2] = blueText.text.toString().toInt().coerceIn(0, 255)
+                val calculatedColorState = ColorUtils.calculateColorWithRGB(rgb)
+                forceViewUpdates(calculatedColorState)
+            }
+        }
+        greenText.addTextChangedListener {
+            if (isIntegerInput(it.toString())) {
+                val rgb = IntArray(3)
+                rgb[0] = redText.text.toString().toInt().coerceIn(0, 255)
+                rgb[1] = it.toString().toInt().coerceIn(0, 255)
+                rgb[2] = blueText.text.toString().toInt().coerceIn(0, 255)
+                val calculatedColorState = ColorUtils.calculateColorWithRGB(rgb)
+                forceViewUpdates(calculatedColorState)
+            }
+        }
+        blueText.addTextChangedListener {
+            if (isIntegerInput(it.toString())) {
+                val rgb = IntArray(3)
+                rgb[0] = redText.text.toString().toInt().coerceIn(0, 255)
+                rgb[1] = greenText.text.toString().toInt().coerceIn(0, 255)
+                rgb[2] = it.toString().toInt().coerceIn(0, 255)
+                val calculatedColorState = ColorUtils.calculateColorWithRGB(rgb)
+                forceViewUpdates(calculatedColorState)
+            }
+        }
+        sheet.setContentView(view)
+        val behavior = BottomSheetBehavior.from(view.parent as View)
+        behavior.state = BottomSheetBehavior.STATE_EXPANDED
+        behavior.isDraggable = false
```

Comment:
should only do this when the gradient or slider are pressed because dragging the sheet down is a valid way of canceling that people like to use

Reply:
Agreed, Will Implement this

Reply:
Implemented